ttnn/cpp/ttnn/operations/moreh/moreh_fold/device/fold_program_factory_rm.cpp

```diff
@@ -0,0 +1,217 @@
+    uint32_t num_cores_x = compute_with_storage_grid_size.x;
+    uint32_t num_cores_y = compute_with_storage_grid_size.y;
+    auto [num_cores, all_cores, core_group_1, core_group_2, num_units_per_core_group_1, num_units_per_core_group_2] =
+        split_work_to_cores(compute_with_storage_grid_size, num_units);
+
+    auto arch = input.device()->arch();
+    auto [math_fidelity, math_approx_mode, fp32_dest_acc_en, packer_l1_acc, dst_full_sync_en] =
+        get_compute_kernel_config_args(arch, compute_kernel_config);
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                         CircularBuffer Setup
+    ////////////////////////////////////////////////////////////////////////////
+    auto data_format = tt::tt_metal::datatype_to_dataformat_converter(input.get_dtype());
+
+    uint32_t unit_size = input.element_size();
+    uint32_t input_cb_page_size = unit_size * input.get_shape()[-1];
+    uint32_t output_cb_page_size = unit_size * output.get_shape()[-1];
+
+    uint32_t aligned_input_cb_page_size = round_up_to_mul32(input_cb_page_size);
+    uint32_t aligned_output_cb_page_size = round_up_to_mul32(output_cb_page_size);
+
+    uint32_t input_cb_index = tt::CB::c_in0;    // input
+    uint32_t output_cb_index = tt::CB::c_out0;  // ouput
+
+    CircularBufferConfig input_cb_config =
+        CircularBufferConfig(aligned_input_cb_page_size * 2, {{input_cb_index, data_format}})
+            .set_page_size(input_cb_index, aligned_input_cb_page_size);
+    auto input_cb = CreateCircularBuffer(program, all_cores, input_cb_config);
+
+    CircularBufferConfig output_cb_config =
+        CircularBufferConfig(aligned_output_cb_page_size * 2, {{output_cb_index, data_format}})
+            .set_page_size(output_cb_index, aligned_output_cb_page_size);
+    auto output_cb = CreateCircularBuffer(program, all_cores, output_cb_config);
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                         Kernels defines
+    ////////////////////////////////////////////////////////////////////////////
+    std::map<string, string> reader_defines;
+    std::map<string, string> writer_defines;
+
+    switch (input.get_dtype()) {
+        case DataType::BFLOAT16: reader_defines["DTYPE_BFLOAT16"] = "1"; break;
+        case DataType::FLOAT32: reader_defines["DTYPE_FLOAT32"] = "1"; break;
+        default: break;
+    }
+
+    if (fp32_dest_acc_en) {
+        reader_defines["FP32_DEST_ACC_EN"] = 1;
+        writer_defines["FP32_DEST_ACC_EN"] = 1;
+    }
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      DataMovementKernel SetUp
+    ////////////////////////////////////////////////////////////////////////////
+    bool src_is_dram = input.buffer()->buffer_type() == tt::tt_metal::BufferType::DRAM ? 1 : 0;
+    bool out_is_dram = output.buffer()->buffer_type() == tt::tt_metal::BufferType::DRAM ? 1 : 0;
```

Comment:
'src' vs 'dst' or 'input' vs 'output' seems more reasonable. In this case, I'd prefer to use 'input' vs 'output.'

Reply:
Resolved in last commit.